File: Ejercicios Propuestos 3/Ejercicio 18/ejercicio18.cpp
#include <iostream>
using namespace std;

int main() {
    int n, m;
    cout << "Ingrese numero de filas: ";
    cin >> n;
    cout << "Ingrese numero de columnas: ";
    cin >> m;

    // Reservar memoria para n filas
    int** M = new int*[n];
    for (int i = 0; i < n; i++) {
        M[i] = new int[m];
    }

    // Llenar la matriz
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < m; j++) {
            cout << "M[" << i << "][" << j << "]: ";
            cin >> M[i][j];
        }
    }

    // Mostrar la matriz
    cout << "Matriz ingresada:" << endl;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < m; j++) {
            cout << M[i][j] << " ";
        }
        cout << endl;
    }

    // Liberar memoria
    for (int i = 0; i < n; i++) {
        delete[] M[i];
    }
    delete[] M;

    return 0;
}

File: Ejercicios Propuestos 3/Ejercicio 18/Ejercicio 18.py
NAME = "Renzo Leonardo Cienfuegos Cardenas"
GROUP = 2

def crear_matriz_dinamica(filas, columnas):
    nueva_matriz = []

    for f in range(filas):
        fila = []
        for c in range(columnas):
            fila.append(0)
        nueva_matriz.append(fila)

    return nueva_matriz


def liberar_matriz_dinamica(matriz):
    del matriz


# Programa principal
n_filas = int(input("Ingrese el número de filas de la matriz: "))
n_columnas = int(input("Ingrese el número de columnas de la matriz: "))

mi_matriz = crear_matriz_dinamica(n_filas, n_columnas)

for f in range(n_filas):
    for c in range(n_columnas):
        mi_matriz[f][c] = int(input(f"Ingrese valor para [{f}][{c}]: "))

print("--------------------------------------------------")
print("CONTENIDO DE LA MATRIZ DINÁMICA:")

for f in range(n_filas):
    for c in range(n_columnas):
        print(mi_matriz[f][c], end=" ")
    print()

print("--------------------------------------------------")

liberar_matriz_dinamica(mi_matriz)
print("Memoria dinámica liberada correctamente.")
